certificate-service/integration-certificate-print-service/src/main/java/se/inera/intyg/certificateservice/certificate/integration/PrintCertificateFromCertificatePrintService.java

```diff
@@ -0,0 +1,49 @@
+package se.inera.intyg.certificateservice.certificate.integration;
+
+import static se.inera.intyg.certificateservice.logging.MdcHelper.LOG_SESSION_ID_HEADER;
+import static se.inera.intyg.certificateservice.logging.MdcHelper.LOG_TRACE_ID_HEADER;
+import static se.inera.intyg.certificateservice.logging.MdcLogConstants.EVENT_TYPE_ACCESSED;
+import static se.inera.intyg.certificateservice.logging.MdcLogConstants.SESSION_ID_KEY;
+import static se.inera.intyg.certificateservice.logging.MdcLogConstants.TRACE_ID_KEY;
+
+import lombok.RequiredArgsConstructor;
+import org.slf4j.MDC;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestClient;
+import se.inera.intyg.certificateservice.certificate.dto.PrintCertificateRequestDTO;
+import se.inera.intyg.certificateservice.certificate.dto.PrintCertificateResponseDTO;
+import se.inera.intyg.certificateservice.logging.PerformanceLogging;
+
+@Service
+@RequiredArgsConstructor
+public class PrintCertificateFromCertificatePrintService {
+
+  private final RestClient restClient;
+
+  @Value("${integration.certificateprintservice.address}")
+  private String printCertificateServiceUrl;
+
+
+  @PerformanceLogging(eventAction = "print-certificate-from-certificate-print-service", eventType = EVENT_TYPE_ACCESSED)
+  public PrintCertificateResponseDTO print(PrintCertificateRequestDTO request,
+      String certificateId) {
+    try {
+      return restClient
+          .post()
+          .uri(printCertificateServiceUrl + "/api/print/" + certificateId)
+          .header(LOG_TRACE_ID_HEADER, MDC.get(TRACE_ID_KEY))
+          .header(LOG_SESSION_ID_HEADER, MDC.get(SESSION_ID_KEY))
+          .body(request)
+          .retrieve()
+          .body(PrintCertificateResponseDTO.class);
+    } catch (HttpClientErrorException ex) {
```

Comment:
Do we need to catch? Why return null if status is NOT_FOUND?

Reply:
used same pattern as intygsadmin but can update

Reply:
I don't think we should return null, suggest that we don't catch any exceptions.